src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -33,6 +33,7 @@ internal sealed class ProjectInterpretation
         private const string PlatformMetadataName = "Platform";
         private const string PlatformsMetadataName = "Platforms";
         private const string EnableDynamicPlatformResolutionMetadataName = "EnableDynamicPlatformResolution";
+        private const string OverridePlatformNegotiationValue = "OverridePlatformNegotiationValue";
```

Comment:
Why a new metadatum rather than using the existing `SetPlatform`?

Reply:
using the set platform metadatum would turn off this feature entirely which we don't want.

Reply:
I think what I was envisioning was:

1. When `EnableDynamicPlatformResolution`
2. And SetPlatform
3. And the value of setplatform is the default for the referenced project
4. then unset the global property instead of setting it explicitly

But based on some conversations offline, that would be a difference from the current behavior, so this is a bit more conservative.

Reply:
for this feature A doesnt need to know the default of B. it simply sets the platform it wants it to build and then negotiation is what decides if that value should be set OR if it should be left blank(in the case the value matches the default). if default of b changes no change is needed from the referencing project.